src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Build {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	public sealed class MandatoryReferencesAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create( Diagnostics.MustReferenceAnnotations );
+
+		public override void Initialize( AnalysisContext ctx ) {
+			ctx.EnableConcurrentExecution();
+			ctx.RegisterCompilationAction( AnalyzeCompilation );
+		}
+
+		public static void AnalyzeCompilation(
+			CompilationAnalysisContext ctx
+		) {
+			bool hasAnnotationsReference = ctx.Compilation
+				.References
+				.Any( IsTheAnnotationsAssembly );
+
+			if ( !hasAnnotationsReference ) {
+				ctx.ReportDiagnostic(
+					Diagnostic.Create(
+						Diagnostics.MustReferenceAnnotations,
+						Location.None
+					)
+				);
+			}
+		}
+
+		public static bool IsTheAnnotationsAssembly( MetadataReference mr ) {
+			return mr.Display.EndsWith( "\\D2L.CodeStyle.Annotations.dll" );

```

Comment:
This isn't fantastic. We could maybe investigate strong-naming the assembly.

Reply:
If this is a path, Consider using Path.GetFileName then.

Reply:
Hm, so the docs for `Display` say its a "path or name". I'm not really sure what that means but I've obviously assumed it's a path. I should probably figure out what's actually happening.